onnxscript/irbuilder.py

```diff
@@ -463,12 +466,23 @@ def add_input(
         v = IRVar(varname, type, info)
         fn.append_input(v)
 
-    def add_attr_parameter(self, fn: IRFunction, varname: str, default_value) -> None:
+    def add_attr_parameter(
+        self,
+        fn: IRFunction,
+        varname: str,
+        attribute_type: onnx.AttributeProto.AttributeType,
+        default_value: Any,
+    ) -> None:
         if default_value is not None:
-            a = IRAttributeValue(helper.make_attribute(varname, default_value))
-            fn.add_attr_parameter(a)
+            fn.add_attr_parameter(
+                IRAttributeValue(helper.make_attribute(varname, default_value)),
+                has_default=True,
+            )
         else:
-            fn.add_attr_parameter(varname)
+            proto = onnx.AttributeProto()
+            proto.name = varname
+            proto.type = attribute_type
+            fn.add_attr_parameter(IRAttributeValue(proto), has_default=False)
```

Comment:
Do you think has_defualt can be an attribute of IRAttributeValue? I think that's more straightforward.

Reply:
That's a great suggestion - I thought of this too. Will make an update